src/api.rs

```diff
@@ -150,8 +157,12 @@ pub async fn blob(req: web::Path<BlobRequest>, qstr: web::Query<ManifestQueryStr
 		return Err(Error::InvalidDigest);
 	}
 
-	let namespace = qstr.ns.as_deref().unwrap_or_else(|| req.image.as_ref().split("/").next().unwrap_or_else(|| config.default_ns.as_ref()));
+	let mut namespace = qstr.ns.as_deref().unwrap_or_else(|| config.default_ns.as_ref());
+	if req.image.as_ref().split("/").count() > 2 {
+		namespace = req.image.as_ref().split("/").next().unwrap_or_else(|| config.default_ns.as_ref());
+	}
 
+	let image: &str = req.image.as_ref().trim_start_matches("docker.io/");
```

Comment:
Same as above here - in fact maybe makes sense to deduplicate that in a helper function.  I don't feel strongly about that though, whatever you feel like doing is fine